src/sentry/models/event.py

```diff
@@ -130,6 +134,14 @@ def error(self):
 
     error.short_description = _('error')
 
+    @property
+    def real_message(self):
+        # XXX(mitsuhiko): this is a transitional attribute that should be
+        # removed.  `message` will be renamed to `search_message` and this
+        # will become `message`.
+        msg_interface = self.data.get('logentry')
+        return msg_interface and (msg_interface.get('formatted') or msg_interface['message']) or ''
```

Comment:
@mitsuhiko could you explain why it falls back to an empty string instead of `self.message` (as it does in `get_legacy_message()`)?
We have recently upgraded our Sentry from 9.0.0 to 9.1.2 and now the alerting system doesn't work when the rule is based on the message text matching. Looks like it might be a bug here.